### PR TITLE
Add `onchainunspent` RPC call

### PR DESCRIPTION
**Pull request by rorp**
This PR add an RPC call to retrieve the list of UTXOs from the connected Bitcoin Core instance.

### CODE REVIEW

**Comment by sstone** [comment]
What is the rationale for adding this to eclair's API ? Users could simply call bitcoin core's API instead ?

File: docs/release-notes/eclair-vnext.md [hunk 1]
```diff
@@ -94,6 +94,7 @@ upgrading to this release.
 - `channelbalances`: retrieves information about the balances of all local channels (#2196)
 - `stop`: stops eclair. Please note that the recommended way of stopping eclair is simply to kill its process (#2233)
 - `channelbalances` and `usablebalances` return a `shortIds` object instead of a single `shortChannelId` (#2323)
+- `onchainunspent`: retrieves the list of UTXOs (#2244)
```
Comment (t-bast) [inline docs/release-notes/eclair-vnext.md:97]:
```suggestion
- `onchainunspent`: retrieves the list of UTXOs (#2344)
```

**Review by t-bast — COMMENTED** [review]
I don't mind adding this RPC, but are you sure this will provide what you want?
An application using this RPC will need to gracefully handle a few edge cases, among which:

- the utxo it selects is actually locked because `eclair` is in the process of using it for a transaction
- the utxo it selects is unlocked, but `eclair` also wants to use it in parallel and creates a conflicting transaction

The only way to work around this would be to expose `bitcoind`'s locking RPCs, which opens a real can of worms we don't want to get into.

**Comment by t-bast** [comment]
@rorp can you reply to the previous comments? Otherwise I'll close this PR as inactive.

**Comment by rorp** [comment]
I agree that it should differentiate between locked and unlocked UTXO.

As @sstone pointed out, all this can be done using Bitcoin Core RPC directly, so the can of worm is already wide open.
 
The lock/unlock UTXO API would be beneficial, because it could perform additional checks. For example, like LND's `leaseOutput`/`releaseOutput` require an app id, so the app cannot unlock outputs locked by Eclair or other apps.

**Comment by t-bast** [comment]
> For example, like LND's leaseOutput/releaseOutput require an app id, so the app cannot unlock outputs locked by Eclair or other apps.

That can only be done when implementing your own bitcoin wallet. Since eclair uses bitcoind instead of re-implementing a wallet, this is impossible for us. That's why I think we simply should not go down that road, because we can't provide the necessary guarantees to the caller.

This should be done by querying bitcoind directly, I don't think we should add even more APIs to eclair for that.

**Comment by t-bast** [comment]
Shall we close this PR? I'm not clear on what the status is here @rorp

**Pull request closed without merging**